dining/views.py

```diff
@@ -16,6 +14,462 @@
+        for preference in aggregated_preferences:
+            venue = Venue.objects.get(id=preference["venue_id"])
+            preference["venue_id"] = venue.venue_id
+
+        return Response({"preferences": aggregated_preferences})
+
+    def post(self, request):
+
+        profile = request.user.profile
+
+        # clears all previous preferences associated with the profile
+        preferences = DiningPreference.objects.filter(profile=profile)
+        for preference in preferences:
+            preference.delete()
+
+        venue_ids = request.data["venues"]
+
+        for venue_id in venue_ids:
+            venue = get_object_or_404(Venue, venue_id=int(venue_id))
+            # adds all of the preferences given by the request
+            DiningPreference.objects.create(profile=profile, venue=venue)
+
+        return Response({"success": True, "error": None})
+
+
+class Transactions(APIView):
+
+    permission_classes = [IsAuthenticated]
+
+    def get(self, request):
+        profile = request.user.profile
+        transactions = DiningTransaction.objects.filter(profile=profile)
+        serializer = DiningTransactionSerializer(transactions, many=True)
+        return Response(serializer.data)
+
+    def post(self, request):
+
+        profile = request.user.profile
+
+        transaction = request.data["transactions"]
+
+        # create list of rows, remove headers, and reverse so in order of date
+        cr = csv.reader(transaction.splitlines(), delimiter=",")
+        row_list = list(cr)
+        row_list.pop(0)
+        row_list.reverse()
+
+        last_transaction = DiningTransaction.objects.order_by("-date").first()
+
+        for row in row_list:
+            if len(row) == 4:
+                if row[0] == "No transaction history found for this date range.":
+                    continue
+                else:
+                    date = datetime.datetime.strptime(row[0], "%m/%d/%Y %I:%M%p")
+                    if last_transaction is None or date > last_transaction.date:
+                        DiningTransaction.objects.create(
+                            profile=profile,
+                            date=date,
+                            description=row[1],
+                            amount=float(row[2], balance=float(row[3])),
+                        )
+
+        return Response({"success": True, "error": None})
+
+
+class Balance(APIView):
+
+    permission_classes = [IsAuthenticated]
+
+    def get_dst_gmt_timezone(self):
+        now = datetime.datetime.now(tz=pytz.timezone("US/Eastern"))
+        if now.timetuple().tm_isdst:
+            return "04:00"
+        else:
+            return "05:00"
+
+    def get(self, request):
+
+        profile = request.user.profile
+        balance = DiningBalance.objects.filter(profile=profile).order_by("-date").first()
+
+        if balance is not None:
+            timestamp = balance.date.strftime("%Y-%m-%dT%H:%M:%S") + "-{}".format(
```

Comment:
I know this was a hack in LAS to start with, but I'm pretty sure we don't need the DST fix since django is timezone aware by default